### PR TITLE
Aligned kedro-viz telemetry with kedro-telemetry

### LINKED ISSUE #2020: Update kedro-viz telemetry to reflect recent changes in kedro-telemetry
## Description
We switched to an opt-out model in Kedro-Telemetry as described in [issue #715](https://github.com/kedro-org/kedro-plugins/issues/715). This change introduces new methods for obtaining telemetry consent, which should be incorporated into the Viz telemetry. Additionally, a new way to establish the user ID was introduced in [PR #596](https://github.com/kedro-org/kedro-plugins/pull/596), and this should also be updated in Viz.

## Context
Why is this change important to you? How would you use it? How can it benefit other users?

## Possible Implementation
(Optional) Suggest an idea for implementing the addition or change.

## Possible Alternatives
(Optional) Describe any alternative solutions or features you've considered.

## Checklist

- [ ] Include labels so that we can categorise your feature request

### PR DESCRIPTION
**Pull request by jitu5**
## Description

Resolves #2020 

This PR updates existing telemetry setup in Kedro-Viz:

- Added `kedro-telemetry` as required dependency. 
- Used `_check_for_telemetry_consent` from `kedro-telemetry`
   - To check environment variables `DO_NOT_TRACK` or `KEDRO_DISABLE_TELEMETRY`
   - To check for consent in `.telemetry` file
- Updated `get_heap_identity` to generate an unique UUID for each user of `kedro-telemetry`.
- If `kedro-telemetry` is not installed, it should be treated as consent being `False`.

**Heap event data:** (Identity)
Before:
Screenshot 2024-09-25 at 10 53 46 a m

After:
Screenshot 2024-09-25 at 11 17 17 a m

## QA notes

- All test should pass
- Screenshot 2024-09-25 at 7 30 10 p m

## Checklist

- [X] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [X] Updated the documentation to reflect the code changes
- [X] Added new entries to the `RELEASE.md` file
- [X] Added tests to cover my changes

### CODE REVIEW

File: package/kedro_viz/integrations/kedro/telemetry.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
-import yaml
-
 try:
-    from kedro_telemetry.plugin import _get_heap_app_id, _is_valid_syntax
+    from kedro_telemetry.plugin import (
+        _check_for_telemetry_consent,
+        _get_heap_app_id,
+        _get_or_create_uuid,
+    )
```
Comment (ravi-kumar-pilla) [inline package/kedro_viz/integrations/kedro/telemetry.py:13]: does this mean, next version of viz should have the latest kedro telemetry to be installed for tracking or is there a way to get latest viz support older kedro telemetry ?

**Review by DimedS — APPROVED** [review]
Thank you @jitu5, LGTM!

**Review by ravi-kumar-pilla — APPROVED** [review]
Code looks good to me 💯 ... thank you @jitu5

**Review by rashidakanchwala — APPROVED** [review]
thank you @jitu5 , Will have to connect with @merelcht  to see if it makes sense to convert the functions to public since Kedro-viz accessed it.

**Pull request merged**